src/Operators/remapping.jl

```diff
@@ -65,116 +66,50 @@ Computes local weights of the overlap mesh for `source` to `target` spaces.
-    S <: SpectralElementSpace1D{<:IntervalTopology, Quadratures.GL{1}},
-}
-    # Calculate element overlap pattern
-    # X_ov[i,j] = overlap length between target elem i and source elem j
-    X_ov = fv_x_overlap(target, source)
-
-    return X_ov
-end
-
-function overlap_weights(
-    target::T,
-    source::S,
 ) where {
     T <: SpectralElementSpace1D{<:IntervalTopology},
     S <: SpectralElementSpace1D{<:IntervalTopology},
 }
-    FT = Spaces.undertype(source)
-    target_topo = Spaces.topology(target)
-    source_topo = Spaces.topology(source)
-    nelems_t = Topologies.nlocalelems(target)
-    nelems_s = Topologies.nlocalelems(source)
-    QS_t = Spaces.quadrature_style(target)
-    QS_s = Spaces.quadrature_style(source)
-    Nq_t = Quadratures.degrees_of_freedom(QS_t)
-    Nq_s = Quadratures.degrees_of_freedom(QS_s)
-    J_ov = spzeros(nelems_t * Nq_t, nelems_s * Nq_s)
-
-    # Calculate element overlap pattern
-    # X_ov[i,j] = overlap length between target elem i and source elem j
-    for i in 1:nelems_t
-        vertices_i = Topologies.vertex_coordinates(target_topo, i)
-        min_i, max_i = Geometry.component(vertices_i[1], 1),
-        Geometry.component(vertices_i[2], 1)
-        for j in 1:nelems_s # elems coincide w nodes in FV source
-            vertices_j = Topologies.vertex_coordinates(source_topo, j)
-            # get interval for quadrature
-            min_j, max_j = Geometry.component(vertices_j[1], 1),
-            Geometry.component(vertices_j[2], 1)
-            min_ov, max_ov = max(min_i, min_j), min(max_i, max_j)
-            if max_ov <= min_ov
-                continue
-            end
-
-            # use smaller order for overlap quadrature
-            ξ_s, w_s = Quadratures.quadrature_points(FT, QS_s)
-            ξ_t, w_t = Quadratures.quadrature_points(FT, QS_t)
-            x_ov =
-                FT(0.5) * (min_ov + max_ov) .+ FT(0.5) * (max_ov - min_ov) * ξ_t
-            w = w_t
-            x_t = FT(0.5) * (min_i + max_i) .+ FT(0.5) * (max_i - min_i) * ξ_t
-            x_s = FT(0.5) * (min_j + max_j) .+ FT(0.5) * (max_j - min_j) * ξ_s
-
-            # column k of I_mat gives the k-th target basis function defined on the overlap element
-            I_mat_t = Quadratures.interpolation_matrix(x_ov, x_t)
-            I_mat_s = Quadratures.interpolation_matrix(x_ov, x_s)
-
-            for k in 1:Nq_t
-                targ_idx = Nq_t * (i - 1) + k # global nodal index
-                for l in 1:Nq_s
-                    src_idx = Nq_s * (j - 1) + l
-                    # (integral of src_basis * tgt_basis on overlap) / (reference elem length * overlap elem length)
-                    J_ov[targ_idx, src_idx] =
-                        (w' * (I_mat_t[:, k] .* I_mat_s[:, l])) ./ 2 *
-                        (max_ov - min_ov)
-                end
-            end
-        end
-    end
-    return J_ov
+    return x_overlap(target, source)
```

Comment:
I think the convention is to use `!` if the function modifies its argument. Is this what this function does to `target`?

Reply:
Please document these functions

Reply:
> I think the convention is to use `!` if the function modifies its argument. Is this what this function does to `target`?

No, the `target` space is not modified. `x_overlap` generates a sparse matrix. I'll be sure to add some docs!